test/proxy.test.ts

```diff
@@ -0,0 +1,83 @@
+import { StarknetContract, StarknetContractFactory, Account } from 'hardhat/types';
+import {starknet} from 'hardhat';
+import {TIMEOUT} from './constants';
+import {expect} from 'chai';
+
+describe('Proxy', function () {
+  this.timeout(TIMEOUT);
+
+  let TokenContractFactory: StarknetContractFactory;
+  let ProxyFactory: StarknetContractFactory;
+  let tokenContractA: StarknetContract;
+  let tokenContractB: StarknetContract;
+  let proxyTokenContract: StarknetContract;
+  let proxiedTokenContract: StarknetContract;
+  let owner: Account;
+  let randomUser: Account;
+
+  before(async () => {
+    owner = await starknet.deployAccount('OpenZeppelin');
+    randomUser = await starknet.deployAccount('OpenZeppelin');
+    TokenContractFactory = await starknet.getContractFactory('ETHstaticAToken');
+    ProxyFactory = await starknet.getContractFactory('proxy');
+
+    tokenContractA = await TokenContractFactory.deploy();
+    tokenContractB = await TokenContractFactory.deploy();
+
+    proxyTokenContract = await ProxyFactory.deploy({ proxy_admin: owner.starknetContract.address });
+
+    await owner.invoke(proxyTokenContract, 'initialize_proxy', { implementation_address: BigInt(tokenContractA.address) });
+    proxiedTokenContract = TokenContractFactory.getContractAt(proxyTokenContract.address);
+
+  });
+
+  it('Verify that owner is the proxy admin', async () => {
+    const { admin } = await proxyTokenContract.call('get_admin', {});
+    expect(admin).to.equal(BigInt(owner.starknetContract.address));
+  });
+
+  it('initialize token A through proxy', async () => {
+    await owner.invoke(proxiedTokenContract, 'initialize_ETHstaticAToken', {
+      name: 666n,
+      symbol: 666n,
+      decimals: 4n,
+      initial_supply: {high: 0n, low: 0n},
+      recipient: BigInt(owner.starknetContract.address),
+      controller: BigInt(owner.starknetContract.address),
+    });
+
+    const { name } = await owner.call(proxiedTokenContract, 'name');
+    expect(name).to.equal(666n);
+
+    const { symbol } = await owner.call(proxiedTokenContract, 'symbol');
+    expect(symbol).to.equal(666n);
+
+    const { decimals } = await owner.call(proxiedTokenContract, 'decimals');
+    expect(decimals).to.equal(4n);
+  });
+
+  it('diasllows random user to upgrade',async () => {
+    try {
+      randomUser.invoke(proxyTokenContract, 'upgrade_implementation',
+          { new_implementation: BigInt(tokenContractB.address) }
+      );
+      expect.fail("Random user cannot upgrade");
```

Comment:
This line keeps on failing tests ` AssertionError: expected 'Random user cannot upgrade' to include 'Proxy: caller is not admin'`. Handling the exception in the catch block should be enough.

Reply:
I think its for better logging, but you're right its redundant

Reply:
I think its for logging purposes, you'r right about it being redundant